projects/bp-gallery/src/components/views/search/SearchView.tsx

```diff
@@ -47,25 +44,26 @@ const SearchView = () => {
 
   // Builds query from search params in the path
   const queryParams = useMemo(() => {
-    if (isAllSearchActive) {
-      const allSearchTerms = searchParams
-        .getAll(toURLSearchParam(SearchType.ALL))
-        .map(decodeURIComponent);
-      const searchTimes: string[][] = [];
-      allSearchTerms.forEach(searchTerm => {
-        if (isValidTimeSpecification(searchTerm)) {
-          const { startTime, endTime } = paramToTime(searchTerm);
-          searchTimes.push([searchTerm, startTime, endTime]);
-        }
-      });
-      return {
-        searchTerms: allSearchTerms.filter(searchTerm => !isValidTimeSpecification(searchTerm)),
-        searchTimes,
-      };
-    }
-    return convertSearchParamsToPictureFilters(searchParams);
-  }, [isAllSearchActive, searchParams]);
-
+    // if (isAllSearchActive) {
```

Comment:
Is there a reason to remove the check?

Reply:
yes, i removed it, because the two returns of the function return different types. By making sure only the first return wil lbe used I could narrow down the type and didn't have to deal with type script bs, that didn't even matter in this case, since we never use allSearch